instrumentation/github.com/gin-gonic/gin/otelgin/gintrace_test.go

```diff
@@ -105,3 +108,39 @@ func TestPropagationWithCustomPropagators(t *testing.T) {
 
 	router.ServeHTTP(w, r)
 }
+
+func TestWithCustomSpanOptions(t *testing.T) {
+	spanRecorder := tracetest.NewSpanRecorder()
+	provider := sdktrace.NewTracerProvider(sdktrace.WithSpanProcessor(spanRecorder))
+	b3 := b3prop.New()
+
+	r := httptest.NewRequest("GET", "/user/123", nil)
+	w := httptest.NewRecorder()
+
+	attrs := trace.WithAttributes(
+		[]attribute.KeyValue{
+			attribute.String("foo", "bar"),
+		}...,
+	)
+
+	router := gin.New()
+	router.Use(Middleware("foobar", WithTracerProvider(provider), WithPropagators(b3), WithSpanOptions(attrs)))
```

Comment:
It doesn't feel like propagators are need for the test to pass?

Reply:
fixed it